CMake/coverage.cmake

```diff
@@ -35,8 +35,46 @@ function(add_converage module)
         add_custom_command(TARGET ${module}-ccov POST_BUILD
             COMMENT "Open ${module}-llvm-cov/index.html in your browser to view the coverage report."
         )
+    elseif("${CMAKE_C_COMPILER_ID}" MATCHES "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
+        message("[Coverage] Building with Gcc Code Coverage Tools")
+
+        find_program(GCOV_PATH gcov)
+        IF(NOT GCOV_PATH)
```

Comment:
While not very important, consider not mixing `ALL_CAPS` and `all_lower` styles in one file?